sdk/lib/opentelemetry/sdk/trace/tracer.rb

```diff
@@ -37,14 +37,14 @@ def start_span(name, with_parent: nil, attributes: nil, links: nil, start_timest
           name ||= 'empty'
 
           with_parent ||= Context.current
-          parent_span_context = OpenTelemetry::Trace.current_span(with_parent).context
-          parent_span_context = nil unless parent_span_context.valid?
-          parent_span_id = parent_span_context&.span_id
-          tracestate = parent_span_context&.tracestate
-          trace_id = parent_span_context&.trace_id
+          parent_span_reference = OpenTelemetry::Trace.current_span(with_parent).reference
+          parent_span_reference = nil unless parent_span_reference.valid?
+          parent_span_id = parent_span_reference&.span_id
+          tracestate = parent_span_reference&.tracestate
+          trace_id = parent_span_reference&.trace_id
```

Comment:
This logic should be cleaned up - I suspect the obvious change would be more efficient than the repeated "lonely operator".